lib/nindo/feeds/sources.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Nindo.Sources do
+  @moduledoc false
+
+  alias NinDB.{Source}
+  alias Nindo.{Accounts, Post}
+
+  import Nindo.Core
+
+  def add(source, user) do
+    sources = user.sources
+    if source not in sources do
+      Accounts.change(:sources, sources ++ [source], user)
+    end
+  end
+
+  def remove(source, user) do
+    sources = user.sources
+    if source in sources do
+      Accounts.change(:sources, sources -- [source], user)
+    end
+  end
+
+  def generate(title, type, url) do
+    %Source{
+      title: title,
+      id: :erlang.phash2(url),
+      feed: url,
+      type: type,
+      icon: "https://#{url}/favicon.ico"
+    }
+  end
+
+  # we're only generating and caching the first five posts due to performance issues
+  def generate_posts(parsed_feed, source) do
```

Comment:
This should also be moved to a different module, although I'm not sure which one specifically. I'm thinking `Nindo.Posts`

Reply:
No, I'd rather keep it here. It generates a list of posts from a parsed feed, but it does so for a specific source. I think it belongs in the sources module.